transaction-view/src/transaction_meta.rs

```diff
@@ -0,0 +1,394 @@
+use {
+    crate::{
+        address_table_lookup_meta::AddressTableLookupMeta,
+        bytes::advance_offset_for_type,
+        instructions_meta::InstructionsMeta,
+        message_header_meta::{MessageHeaderMeta, TransactionVersion},
+        result::{Result, TransactionParsingError},
+        signature_meta::SignatureMeta,
+        static_account_keys_meta::StaticAccountKeysMeta,
+    },
+    solana_sdk::{hash::Hash, pubkey::Pubkey, signature::Signature},
+};
+
+pub struct TransactionMeta {
+    /// Signature metadata.
+    pub(crate) signature: SignatureMeta,
+    /// Message header metadata.
+    pub(crate) message_header: MessageHeaderMeta,
+    /// Static account keys metadata.
+    pub(crate) static_account_keys: StaticAccountKeysMeta,
+    /// Recent blockhash offset.
+    pub(crate) recent_blockhash_offset: u16,
+    /// Instructions metadata.
+    pub(crate) instructions: InstructionsMeta,
+    /// Address table lookup metadata.
+    pub(crate) address_table_lookup: AddressTableLookupMeta,
+}
+
+impl TransactionMeta {
+    /// Parse a serialized transaction and verify basic structure.
+    /// The `bytes` parameter must have no trailing data.
+    pub fn try_new(bytes: &[u8]) -> Result<Self> {
+        let mut offset = 0;
+        let signature = SignatureMeta::try_new(bytes, &mut offset)?;
+        let message_header = MessageHeaderMeta::try_new(bytes, &mut offset)?;
+        let static_account_keys = StaticAccountKeysMeta::try_new(bytes, &mut offset)?;
+
+        // The recent blockhash is the first account key after the static
+        // account keys. The recent blockhash is always present in a valid
+        // transaction and has a fixed size of 32 bytes.
+        let recent_blockhash_offset = offset as u16;
+        advance_offset_for_type::<Hash>(bytes, &mut offset)?;
+
+        let instructions = InstructionsMeta::try_new(bytes, &mut offset)?;
+        let address_table_lookup = match message_header.version {
+            TransactionVersion::Legacy => AddressTableLookupMeta {
+                num_address_table_lookup: 0,
+                offset: 0,
+            },
```

Comment:
No - because the first thing it does is read the number of ATLs. But that number is only there if the version is V0. In legacy we've already reached the end of the buffer at this point.